src/sessions.ts

```diff
@@ -855,6 +881,10 @@ function updateSessionFromResponse(session: ClientSession, document: Document):
   if (document.recoveryToken && session && session.inTransaction()) {
     session.transaction._recoveryToken = document.recoveryToken;
   }
+
+  if (document.cursor?.atClusterTime && session?.snapshot && session.snapshotTime === undefined) {
+    session.snapshotTime = document.cursor.atClusterTime;
+  }
 }
 
 export {
```

Comment:
If you want to do some legacy clean up, putting the export on each symbol is what we try to do everywhere else in the driver, completely optional.

Reply:
Was easy enough, pushed